src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/grammar/java9/InputAdvanceJava9TryWithResources.java

```diff
@@ -0,0 +1,55 @@
+//non-compiled with javac: Compilable with Java9
+package com.puppycrawl.tools.checkstyle.grammar.java9;
+
+/**
+ * Input for Advance Java 9 try-with-resources.
+ */
+public class InputAdvanceJava9TryWithResources implements AutoCloseable
+{
+    public static void main(String[] args) throws Exception {
+        InputAdvanceJava9TryWithResources v = new InputAdvanceJava9TryWithResources();
+
+        try (v.finalWrapper.finalField) {
+        }
+
+        try (new InputAdvanceJava9TryWithResources() { }.finalWrapper.finalField) {
+        }
+
+        try ((args.length > 0 ? v
+                : new InputAdvanceJava9TryWithResources()).finalWrapper.finalField) {
+        }
+
+        //More than one resource
+        InputAdvanceJava9TryWithResources i1 = new InputAdvanceJava9TryWithResources();
+        try (i1; InputAdvanceJava9TryWithResources i2 =
+                    new InputAdvanceJava9TryWithResources()) {
+        }
+
+        InputAdvanceJava9TryWithResources m1 = new InputAdvanceJava9TryWithResources();
+        try (m1; InputAdvanceJava9TryWithResources m2 = m1;
+                InputAdvanceJava9TryWithResources m3 = m2;) {
+        }
+
+        try {
+        } catch (Exception ex) {
+            try (v) {
```

Comment:
please explain what for we have this code in Input ?
what case you try to cover?

the same question to code below `//More than one resource`

Reply:
This covers the nested try case. If it is not relevant i will remove the code. 
`//More than one resource` for this it covers the case of multiple resources in a try statement.

Reply:
I want only to separate all changes are essential for issue to implement and test AND one changes that good to have but they are not related. All unrelated changes can stay in PR, but should be moved to separate commit `minor: extra test inputs for TRY blocks`.
please remove them or move to separate commit.

Reply:
I removed the code @romani please check